tests/Unit/Http/FosRequestMatcherTest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCacheBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCacheBundle\Tests\Unit\Http;
+
+use FOS\HttpCacheBundle\Http\RequestMatcher\FosRequestMatcher;
+use Symfony\Component\HttpFoundation\Request;
+
+class FosRequestMatcherTest extends \PHPUnit_Framework_TestCase
+{
+    public function testMatchesReturnsFalseIfParentCallFails()
+    {
+        $requestMatcher = new FosRequestMatcher('/foo');
+        $request = Request::create('http://localhost/bar?token=myvalue');
+
+        $this->assertTrue($requestMatcher->matches($request));
+    }
+
+    public function testMatchesReturnsTrueWhenNoQueryStringIsSet()
+    {
+        $requestMatcher = new FosRequestMatcher();
+        $request = Request::create('http://localhost/bar?token=myvalue');
+
+        $this->assertTrue($requestMatcher->matches($request));
+    }
+
+    public function testMatchesReturnsTrueIfQueryStringMatches()
+    {
+        $requestMatcher = new FosRequestMatcher();
+        $requestMatcher->setQueryString('(^|&)token=');
```

Comment:
can we do something here that also changes with the url encoding, to make sure that bit of the code is guaranteed behaviour too?